test/setup_aot/TestSetupAOT.java

```diff
@@ -44,16 +52,161 @@
+            for (File file : files) {
+                deleteAll(file);
+            }
         }
+        System.out.println("Deleting: " + f);
+        f.delete();
+    }
+
+    static void execTool(String tool, String... args) throws Throwable {
+        System.out.println("== Running tool ======================================================");
+        System.out.print(tool);
+        for (String s : args) {
+            System.out.print(" " + s);
+        }
+        System.out.println();
+        System.out.println("======================================================================");
+
+        ToolProvider t = ToolProvider.findFirst(tool)
+            .orElseThrow(() -> new RuntimeException(tool + " not found"));
+        t.run(System.out, System.out, args);
+    }
+
+
+    static void streamOps(String args[]) {
+        List<String> strings = Arrays.asList("Hello", "World!");
+
+        String helloWorld = strings.parallelStream()
+                .filter(s -> s.contains("o"))
+                .map(s -> s.toLowerCase(Locale.ROOT))
+                .collect(joining(","));
+
+        Stream.of(helloWorld.split("([,x-z]{1,3})([\\s]*)"))
+                .map(String::toString)
+                .forEach(System.out::println);
+
+        // Common concatenation patterns
+        int i = args.length;
+        String s = String.valueOf(i);
+
+        String SS     = s + s;
+        String CS     = "string" + s;
+        String SC     = s + "string";
+        String SCS    = s + "string" + s;
+        String CSS    = "string" + s + s;
+        String CSC    = "string" + s + "string";
+        String SSC    = s + s + "string";
+        String CSCS   = "string" + s + "string" + s;
+        String SCSC   = s + "string" + s + "string";
+        String CSCSC  = "string" + s + "string" + s + "string";
+        String SCSCS  = s + "string" + s + "string" + s;
+        String SSCSS  = s + s + "string" + s + s;
+        String S5     = s + s + s + s + s;
+        String S6     = s + s + s + s + s + s;
+        String S7     = s + s + s + s + s + s + s;
+        String S8     = s + s + s + s + s + s + s + s;
+        String S9     = s + s + s + s + s + s + s + s + s;
+        String S10    = s + s + s + s + s + s + s + s + s + s;
+
+        String CI     = "string" + i;
+        String IC     = i + "string";
+        String SI     = s + i;
+        String IS     = i + s;
+        String CIS    = "string" + i + s;
+        String CSCI   = "string" + s + "string" + i;
+        String CIC    = "string" + i + "string";
+        String CICI   = "string" + i + "string" + i;
+
+        float f = 0.1f;
+        String CF     = "string" + f;
+        String CFS    = "string" + f + s;
+        String CSCF   = "string" + s + "string" + f;
+
+        char c = 'a';
+        String CC     = "string" + c;
+        String CCS    = "string" + c + s;
+        String CSCC   = "string" + s + "string" + c;
+
+        long l = System.currentTimeMillis();
```

Comment:
the l should be logged and  allow to be defined using property.
So any testing could be repeated.
 long l =  Long.getLong("test.aot.seed", System.currentTimeMillis());

Reply:
I changed it to not depend on system time. Now the program should be deterministic.